LabelsRandomColors2.0/src/com/company/Main.java

```diff
@@ -46,36 +44,38 @@ private void newLabel(String s) {
         GLabel label = new GLabel(s);
         label.setColor(rgen.nextColor());
         label.setFont("ARIAL-18");
+
         add(label, rgen.nextDouble(0, (getWidth() - label.getWidth())),
                 rgen.nextDouble(label.getAscent(), getHeight() - label.getAscent()));
     }
 
+
     public void mousePressed(MouseEvent e) {
 
         /* The last mouse position */
         GPoint last = new GPoint(e.getX(), e.getY()); //last point on Canvas where mouse was clicked
         gobj = (GLabel) getElementAt(last);

```

Comment:
Potrebbe essere carino se 'nascondessi' i dettagli di come si fa a trovare la label (che ci sia o meno) dietro a una funzione, tipo:

```java
public void mousePressed(MouseEvent event) {

    gobj = findElement(event)
   ...
```

Quando penso a questo aspetto del codice, lo paragono ad una ricetta; ad esempio, questa mi piace:

```
RICETTA TORTA
  prepara gli ingredienti
  mescolali
  cuocili

PREPARA GLI INGREDIENTI
  rompi le uova in una ciotola
  pesa 300g di zucchero

MESCOLALI
  unisci tutti gli ingredienti in una ciotola
  gira con un cucchiaio di legno

CUOCILI
  accendi il forno
  metti gli ingredienti in una teglia in forno
  rimuovi la teglia dopo 30 minuti
```

mentre questa un po meno:
```
RICETTA TORTA
  prepara gli ingredienti
  unisci tutti gli ingredienti in una ciotola
  gira con un cucchiaio di legno
  cuocili

PREPARA GLI INGREDIENTI
  rompi le uova in una ciotola
  pesa 300g di zucchero

CUOCILI
  accendi il forno
  metti gli ingredienti in una teglia in forno
  rimuovi la teglia dopo 30 minuti
```

Questo concetto si puo' chiamare 'Consistent levels of abstraction'. Non ho trovato un buon articolo online, ma la definizione su Wikipedia e' [qui](https://en.wikipedia.org/wiki/Abstraction_(computer_science)#Levels_of_abstraction)

Reply:
THANKS :)